logstash-core/src/main/java/org/logstash/instrument/witness/PluginsWitness.java

```diff
@@ -0,0 +1,99 @@
+package org.logstash.instrument.witness;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * A Witness for the set of plugins.
+ */
+public class PluginsWitness{
+
+    private final Map<String, PluginWitness> inputs;
+    private final Map<String, PluginWitness> outputs;
+    private final Map<String, PluginWitness> filters;
+    private final Forgetter forgetter;
+
+    /**
+     * Constructor.
+     */
+    public PluginsWitness() {
+
+        this.inputs = new HashMap<>();
```

Comment:
We will be accessing these maps across threads right? I'm noticing they aren't CHMs. I think there's a missing piece to the concurrency story here.

I think the best option here would be to have the constructor take the full immutable maps of plugins as arguments, make `forget` in `PipelineWitness` instantiate a brand new `PluginWitness`, and make all references to `PluginsWitness` volatile from `PipelineWitness` to complete the concurrency story.

I wish java actually had a type for `UnmodifiableMap`, but it'd probably be good to pass unmodifiable maps into here to be defensive.

Alternatively, I'm fine just making these CHMs. They aren't expected to be traversed with any frequency.